Source/WebCore/css/parser/CSSParserEnum.h

```diff
@@ -32,7 +32,12 @@ namespace WebCore {
 
 namespace CSSParserEnum {
 
-enum class IsNestedContext : bool { No, Yes };
+enum class AncestorRuleType : bool {
+    Style,
+    Scope,
+};
+
+using IsNestedContext = std::optional<AncestorRuleType>;
```

Comment:
Having typealias called `IsNestedContext` for `std::optional<AncestorRuleType>` is confusing to the point I don't understand what it is trying to say. Can you come up with a better factoring/naming here?